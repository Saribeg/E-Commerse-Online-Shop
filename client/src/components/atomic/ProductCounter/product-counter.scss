@import '../../../scss/resources/resources';

.product-counter{
	display: flex;
	align-items: center;
	border: 1px solid #dedede;
	min-width: 120px;

	&-btn{
		@include clear-default-inputs-style;
		background-color: $grey-bg;
		cursor: pointer;
		font-size: 36px;
		padding: 5px 18px;


	}
	&-value{
		min-width: 45px;
		padding: 5px 18px;
		background-color: #fff;
		font-size: 14px;
	}
}
