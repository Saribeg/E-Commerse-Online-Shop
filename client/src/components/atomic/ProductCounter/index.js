// @flow

import * as React from "react";
import "./product-counter.scss";

type Props = {
  maxCount: number,
};

type State = {
  value: number,
};

export default class ProductCounter extends React.Component<Props, State> {
	state = {
		value: 0,			
	}

	handleChange = (event) => {
    this.setState({value: event.target.value});
  }

	increaseCount = (maxcount) => {
			let existValue = this.state.value;
			if (existValue > (maxcount - 1)){
				return;
			}
			++existValue;
			this.setState({
				value: existValue
			})
	}

	decreaseCount = () => {
		let existValue = this.state.value;
		if( existValue <=0 ){
			return;
		}		
		existValue--;
		this.setState({
			value: existValue
		})
	}
  render() {

		let inputValue = this.state.value;

		if(inputValue > this.props.maxCount){
			this.setState({
				value: this.props.maxCount
			})
		}

		if (typeof inputValue !== 'number'){
			this.setState({
				value: this.props.maxCount
			})
		}
		
    return (
			<div className="product-counter">
			<button className="product-counter-btn" onClick={this.decreaseCount}>-</button>
			<input type="tel" className="product-counter-value" value={this.state.value} onInput={this.handleChange}/>
			<button className="product-counter-btn" onClick={() => this.increaseCount(this.props.maxCount)}>+</button>
	</div>
    );
  }
}
