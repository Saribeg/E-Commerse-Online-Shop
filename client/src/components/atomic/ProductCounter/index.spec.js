import React from 'react';
import ProductCounter from './index';

describe('ProductCounter component', () => {
    it('ProductCounter component render', () => {
        const wrapper  = shallow(<ProductCounter />)
        expect(wrapper).toMatchSnapshot()
    })
})