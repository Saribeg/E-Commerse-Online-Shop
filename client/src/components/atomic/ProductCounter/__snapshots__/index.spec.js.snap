// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCounter component ProductCounter component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductCounter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="product-counter-btn"
          onClick={[Function]}
        >
          -
        </button>,
        <input
          className="product-counter-value"
          onInput={[Function]}
          type="tel"
          value={0}
        />,
        <button
          className="product-counter-btn"
          onClick={[Function]}
        >
          +
        </button>,
      ],
      "className": "product-counter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-",
          "className": "product-counter-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "-",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "product-counter-value",
          "onInput": [Function],
          "type": "tel",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "className": "product-counter-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="product-counter-btn"
            onClick={[Function]}
          >
            -
          </button>,
          <input
            className="product-counter-value"
            onInput={[Function]}
            type="tel"
            value={0}
          />,
          <button
            className="product-counter-btn"
            onClick={[Function]}
          >
            +
          </button>,
        ],
        "className": "product-counter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "className": "product-counter-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "-",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "product-counter-value",
            "onInput": [Function],
            "type": "tel",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "className": "product-counter-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
